src/averageAges.js

```diff
@@ -36,8 +46,20 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
+//  person.mother === withChildren
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women
+    = arguments.length > 1
+      ? people.filter(
+        person =>
+          people.some(child => child.mother === person.name) === withChildren
+          && person.sex === 'f')
+      : people.filter(person => person.sex === 'f');
```

Comment:
```suggestion
  const women = people.filter(person => withChildren
    ? person.sex === 'f'
    : person.sex === 'f' && people.some(child => child.mother === person.name);
```

Reply:
Also why do you need to check arguments length here? Just check second (conditional) argument value